comit/src/hbit.rs

```diff
@@ -221,14 +225,104 @@ where
 
 #[derive(Clone, Copy, Debug)]
 pub struct Params {
-    pub network: bitcoin::Network,
+    pub network: Network,
     pub asset: asset::Bitcoin,
     pub redeem_identity: identity::Bitcoin,
     pub refund_identity: identity::Bitcoin,
     pub expiry: Timestamp,
     pub secret_hash: SecretHash,
 }
 
+impl Params {
+    pub fn build_fund_action(&self) -> SendToAddress {
+        let network = self.network;
+        let to = self.compute_address();
+        let amount = self.asset;
+
+        SendToAddress {
+            to,
+            amount,
+            network,
+        }
+    }
+
+    pub fn build_refund_action<C>(
+        &self,
+        secp: &'static Secp256k1<C>,
+        fund_amount: asset::Bitcoin,
+        fund_location: htlc_location::Bitcoin,
+        transient_refund_sk: SecretKey,
+        refund_address: Address,
+    ) -> anyhow::Result<BroadcastSignedTransaction>
+    where
+        C: Signing,
+    {
+        self.build_spend_action(
+            &secp,
+            fund_amount,
+            fund_location,
+            refund_address,
+            Box::new(move |htlc: BitcoinHtlc| htlc.unlock_after_timeout(secp, transient_refund_sk)),
+        )
+    }
+
+    pub fn build_redeem_action<C>(
+        &self,
+        secp: &'static Secp256k1<C>,
```

Comment:
Does anyone know if we can get rid of this `'static`?

Reply:
I can! No need for dynamic dispatch here (https://users.rust-lang.org/t/solved-understanding-why-a-closure-requires-static-lifetime/23887).